File: app/src/main/java/com/turastory/kotlindemo/todo/app/UsingOverloads.java
package com.turastory.kotlindemo.todo.app;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;

import com.annimon.stream.function.Consumer;
import com.turastory.kotlindemo.stub.InbrainDialogFragment;

/**
 * Created by tura on 2018-09-26.
 */
public class UsingOverloads extends AppCompatActivity {
    
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        
        FragmentManager fm = getSupportFragmentManager();
        
        DialogUtils.message(fm, "asdf");
        DialogUtils.message(fm, "asdf", stub());
        DialogUtils.message(fm, "asdf", 100);
        DialogUtils.message(fm, "asdf", 100, stub());
        
        DialogUtils.skip(fm, dialogStub());
        DialogUtils.skip(fm, dialogStub(), dialogStub());
        
        DialogUtils.backLogin(this, dialogStub());
        DialogUtils.backLogin(this, dialogStub(), 0);
    }
    
    private Runnable stub() {
        return () -> {
        };
    }
    
    private Consumer<? super InbrainDialogFragment> dialogStub() {
        return dialog -> {
        };
    }
}

File: app/src/main/java/com/turastory/kotlindemo/kotlin/app/UsingOverloads.kt
package com.turastory.kotlindemo.kotlin.app

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.annimon.stream.function.Consumer
import com.turastory.kotlindemo.stub.InbrainDialogFragment

/**
 * Created by tura on 2018-09-26.
 */
class UsingOverloads : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val fm = supportFragmentManager

        DialogUtils.message(fm, "asdf")
        DialogUtils.message(fm, "asdf", onDismiss = stub())
        DialogUtils.message(fm, "asdf", dismissTime = 100)
        DialogUtils.message(fm, "asdf", 100, stub())

        DialogUtils.skip(fm, dialogStub())
        DialogUtils.skip(fm, dialogStub(), dialogStub())

        DialogUtils.backLogin(this, dialogStub())
        DialogUtils.backLogin(this, dialogStub(), 0)
    }

    private fun stub(): Runnable {
        return Runnable { }
    }

    private fun dialogStub(): Consumer<in InbrainDialogFragment> {
        return Consumer { }
    }
}
